### PR TITLE
Get interfaces ip

### LINKED ISSUE #204: get_interfaces_ip executes a show command for each interface
And is consequently very slow:

```
        for interface in interfaces:
            show_command = "show run interface {0}".format(interface)
```

Probably replace with `show ip interface`

### PR DESCRIPTION
**Pull request by targuan**
Fix #204
As proposed in the issue by ktbyers I used ```show ip[v6] interface```.

I changed the documentation of get_interfaces_ip in the code as N/A was not accepted on link-local ipv6 prefix_length because it was not an integer. I replaced N/A by 64 which seems to be the prefix length of such addresses.

### CODE REVIEW

File: napalm_ios/ios.py [hunk 1]
```diff
@@ -1038,86 +1038,52 @@ def get_interfaces_ip(self):
...
+            if(line[0] != ' '):
+                ipv4 = {}
+                interfaces[line.split()[0]] = {'ipv4': ipv4}
+            m = re.match(INTERNET_ADDRESS, line)
+            if m:
+                ip, prefix = m.groups()
+                ipv4.update({ip: {"prefix_length": int(prefix)}})
+
+        # Remove interfaces without ip
+        interfaces = dict(filter(lambda x: x[1]['ipv4'] != {}, interfaces.items()))
```
Comment (bewing) [inline napalm_ios/ios.py]: Might this be more readable as
```python
interfaces = {intf: details for intf, details in interfaces.items() if details['ipv4'] != {}}
```
?

**Comment by ktbyers** [comment]
FYI, we agreed on /10 for link-local addresses after reviewing some documents on it:

https://github.com/napalm-automation/napalm-base/issues/304

A bit of ambiguity on whether we should use /64 or /10 for this.

File: napalm_ios/ios.py (hunk 1 above)
Comment (ktbyers) [reply napalm_ios/ios.py]: Yes, I think I would probably change this slightly to something like:

```
            if(line[0] != ' '):
                ipv4 = {}
                interface_name = line.split()[0]
            m = re.match(INTERNET_ADDRESS, line)
            if m:
                ip, prefix = m.groups()
                ipv4.update({ip: {"prefix_length": int(prefix)}})
                interfaces[interface_name] = {'ipv4': ipv4}
```

i.e. only add it to 'interfaces' if there is a positive match

**Comment by ktbyers** [comment]
Only other comment I have is that I need to verify how we handle secondary addresses by default. In other words, whether we do anything to differentiate between a secondary or primary address in the returned data structure.

**Comment by targuan** [comment]
Thanks for your feebacks.
I've updated my PR according to your remarks.

**Comment by ktbyers** [comment]
The secondary address form is correct. So this should be right.

**Pull request merged**